File: ASCOM.Utilities/ASCOM Diagnostics/VersionForm.vb
Imports ASCOM.Utilities
Imports Microsoft.Win32
Imports ASCOM.Utilities.Global

Public Class VersionForm

    Private Sub VersionForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim ProfileKey As RegistryKey

        Try

            Using Profile As New RegistryAccess
                ProfileKey = Profile.OpenSubKey3264(RegistryHive.LocalMachine, "SOFTWARE\ASCOM\Platform", False, RegistryAccess.RegWow64Options.KEY_WOW64_32KEY)

                NameLbl.Text = ProfileKey.GetValue("Platform Name", "Unknown Name")
                Version.Text = ProfileKey.GetValue("Platform Version", "Unknown Version")
                BuildSha.Text = Application.ProductVersion

                ProfileKey.Close()
            End Using

        Catch ex As Exception
            LogEvent("DiagnosticsVersionForm", "Exception", EventLogEntryType.Error, EventLogErrors.DiagnosticsLoadException, ex.ToString)
            MsgBox("Unexpected exception loading Version Form: " & ex.ToString, MsgBoxStyle.Critical, "Version load error")
        End Try

    End Sub
End Class

File: Diagnostics/ASCOM Diagnostics/VersionForm.cs
using System;
using System.Diagnostics;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using Microsoft.Win32;
using static ASCOM.Utilities.Global;
using static ASCOM.Utilities.RegistryAccess;

namespace ASCOM.Utilities
{
    /// <summary>
    /// VersionForm class
    /// </summary>
    public partial class VersionForm
    {
        /// <summary>
        /// VersionForm initiator
        /// </summary>
        public VersionForm()
        {
            InitializeComponent();
        }

        private void VersionForm_Load(object sender, EventArgs e)
        {
            RegistryKey ProfileKey;

            try
            {

                using (var Profile = new RegistryAccess())
                {
                    ProfileKey = Profile.OpenSubKey3264(Registry.LocalMachine, @"SOFTWARE\ASCOM\Platform", false, RegistryAccess.RegistryAccessRights.Wow64_32Key);

                    NameLbl.Text = Conversions.ToString(ProfileKey.GetValue("Platform Name", "Unknown Name"));
                    Version.Text = Conversions.ToString(ProfileKey.GetValue("Platform Version", "Unknown Version"));
                    BuildSha.Text = Application.ProductVersion;

                    ProfileKey.Close();
                }
            }

            catch (Exception ex)
            {
                Utilities.Global.LogEvent("DiagnosticsVersionForm", "Exception", EventLogEntryType.Error, EventLogErrors.DiagnosticsLoadException, ex.ToString());
                Interaction.MsgBox("Unexpected exception loading Version Form: " + ex.ToString(), MsgBoxStyle.Critical, "Version load error");
            }

        }
    }
}